Code/Framework/AzFramework/AzFramework/IO/LocalFileIO.cpp

```diff
@@ -480,22 +488,32 @@ namespace AZ
                 return false;
             }
 
-            if (AZ::IO::PathView(path).HasRootPath())
+            if (AZ::IO::PathView pathView(path);
+                pathView.HasRootPath())
             {
-                size_t pathLen = strlen(path);
-                if (pathLen + 1 < resolvedPathSize)
+                if (pathView.RelativePath().Native().starts_with("@"))
                 {
-                    azstrncpy(resolvedPath, resolvedPathSize, path, pathLen + 1);
-
-                    //see if the absolute path matches the resolved value of @products@, if it does lowercase the relative part
-                    LowerIfBeginsWith(resolvedPath, resolvedPathSize, GetAlias("@products@"));
-
-                    ToUnixSlashes(resolvedPath, resolvedPathSize);
-                    return true;
+                    // If the absolute path starts with an alias then
+                    // remove the root so it can be resolved correctly.
+                    path += AZStd::string_view(path).find_first_of('@');
```

Comment:
since the `pathView.RelativePath` function would return the path part after the root part of the path, you could use that to update the `const char* path` pointer.

My suggestion however is to just iterate over the path and assign the addressd of the last path segment that begins with `@` sign

```
const char* candidatePath = path;
// Iterate over each relative path segment.
// If it starts with an '@' sign, it replaces all previous path segments
// so update the candidatePath to start at that address
for (AZ::IO::PathView pathSegment : pathView.RelativePath())
{
    if (pathView.Native().starts_with('@'))
    {
        candidatePath = pathView.c_str();
    }
}

path = candidatePath;
```

Reply:
That's too much of a change I think it's required for this PR. Looping through the relative path segments finding the last alias is something new that this function has not been doing before. I feel more conformable just removing the root and letting the rest of the function handling it as it has been handling until now. 

Feel free to create another task to improve this function further and do the proper tests about all the possible scenarios. But IMO this shouldn't be blocking this PR from being merged.

 > since the pathView.RelativePath function would return the path part after the root part of the path, you could use that to update the const char* path pointer.
 
I tried that but `PathView` doesn't provide `c_str()`, only Native() and `string_view()` (which is not safe to convert to const char*). I'd have to create an actual path from the view, which would make a copy unnecessarily. If I'm missing something, please let me know. Cheers.

Reply:
The `string_view` function is safe to convert to a `const char*` in this case because the portion of the path we want is from that point where we located the '@' symbol until the end of the original `path` argument. 

Now for the reason for looping over the relative path segments is finding the last path segment, is that this code doesn't work for a path such as "\@product\@/\@projectroot\@/Assets/test.json" or "/\@product\@/\@projectroot\@/\@engroot\@/Registry/bootstrap.setreg"